module-bsp/board/rt1051/puretx/hal/battery_charger/BatteryCharger.cpp

```diff
@@ -1,120 +1,201 @@
 // Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
-#include "BatteryCharger.hpp"
+#include "BatteryChargerIRQ.hpp"
+#include "common/WorkerQueue.hpp"
 
+#include <hal/battery_charger/AbstractBatteryCharger.hpp>
 #include <bsp/battery_charger/battery_charger.hpp>
-#include <hal/GenericFactory.hpp>
 
-namespace hal::battery
-{
-    BatteryCharger::BatteryCharger(AbstractBatteryCharger::BatteryChargerEvents &eventsHandler)
-        : eventsHandler(eventsHandler)
-    {}
+#include <magic_enum.hpp>
 
-    void BatteryCharger::init(xQueueHandle queueBatteryHandle, xQueueHandle queueChargerDetect)
+namespace
+{
+    hal::battery::AbstractBatteryCharger::ChargingStatus transformChargingState(
+        bsp::battery_charger::batteryRetval status)
     {
-        bsp::battery_charger::init(queueBatteryHandle, queueChargerDetect);
+        using Status   = bsp::battery_charger::batteryRetval;
+        using NewState = hal::battery::AbstractBatteryCharger::ChargingStatus;
+        switch (status) {
+        case Status::ChargerCharging:
+            return NewState::Charging;
+        case Status::ChargingDone:
+            return NewState::ChargingDone;
+        case Status::ChargerNotCharging:
+            return NewState ::Discharging;
+        default:
+            return NewState::PluggedNotCharging;
+        }
     }
 
-    void BatteryCharger::BatteryCharger::deinit()
-    {
-        bsp::battery_charger::deinit();
-    }
+    /// A few constants to make code readability better
+    constexpr auto int_b_soc_change =
+        static_cast<std::uint16_t>(bsp::battery_charger::batteryINTBSource::SOCOnePercentChange);
+    constexpr auto int_b_max_temp = static_cast<std::uint16_t>(bsp::battery_charger::batteryINTBSource::maxTemp);
+    constexpr auto int_b_min_temp = static_cast<std::uint16_t>(bsp::battery_charger::batteryINTBSource::minTemp);
+    constexpr auto int_b_all      = static_cast<std::uint16_t>(bsp::battery_charger::batteryINTBSource::all);
+    constexpr auto int_b_miv_v    = static_cast<std::uint16_t>(bsp::battery_charger::batteryINTBSource::minVAlert);
```

Comment:
Minor: rather use camelCase instead of underscores

Reply:
What should it look like? `IntBMaxTemp` or `INTBMaxTemp` or maybe `intBMaxTemp` ? IMHO, in this scenario, the snake case is more readable hence that's why I picked it over camelCase. I always try to choose the right tool for the job instead of following blindly some rules ;)

Reply:
Just pointing out that it is not the style our code is written in ;) For me it helps to see at first glance if it's third party or our code. No strong opinion for that.